src/TabWidget.js

```diff
@@ -0,0 +1,386 @@
+ * @class argos.TabWidget
+ * @alternateClassName TabWidget
+ */
+import declare from 'dojo/_base/declare';
+import lang from 'dojo/_base/lang';
+import array from 'dojo/_base/array';
+import domClass from 'dojo/dom-class';
+import domConstruct from 'dojo/dom-construct';
+import domStyle from 'dojo/dom-style';
+import query from 'dojo/query';
+import _Templated from 'argos/_Templated';
+
+const __class = declare('argos.TabWidget', [_Templated], {
+  /**
+   * @property {Simplate}
+   * HTML that starts a new tab list
+   */
+  tabContentTemplate: new Simplate([
+    '{%! $.tabListTemplate %}',
+  ]),
+  /**
+   * @property {Simplate}
+   * HTML that starts a new tab list
+   */
+  tabListTemplate: new Simplate([
+    '<ul class="tab-list" data-dojo-attach-point="tabList">',
+    '</ul>',
+  ]),
+  /**
+   * @property {Simplate}
+   * HTML that starts a new More tab list
+   */
+  moreTabListTemplate: new Simplate([
+    '<ul class="more-tab-dropdown" data-dojo-attach-point="moreTabList">',
+    '</ul>',
+  ]),
+  /**
+   * @property {Simplate}
+   * HTML that starts a new animation bar
+   */
+  tabListAnimTemplate: new Simplate([
+    '<div class="tab-focus-indicator"></div>',
+    '<div class="animated-bar"></div>',
+  ]),
+  /**
+   * @property {Simplate}
+   * HTML that creates a new tab to be placed in the tab list
+   *
+   * `$` => the view instance
+   */
+  tabListItemTemplate: new Simplate([
+    '<li class="tab" data-action="selectedTab">',
+    '{%: ($.title || $.options.title) %}',
+    '</li>',
+  ]),
+  /**
+   * @property {Simplate}
+   * HTML that creates a new tab to be placed in the more tab list
+   *
+   * `$` => the view instance
+   */
+  moreTabItemTemplate: new Simplate([
+    '<li class="tab more-item" data-action="toggleDropDown">',
+    '{%: ($.title || $.options.title) %}',
+    '<span class="fa fa-angle-right"></span>',
+    '</li>',
+  ]),
+
+  /**
+   * @cfg {String}
+   * More text that is used as the overflow tab for the tab list
+   */
+  moreText: 'More',
+  /**
+   * @property {li}
+   * Current tab (html element li) that the view is on
+   */
+  currentTab: null,
+  /**
+   * @property {bool}
+   * Boolean value for whether tabs caused an overflow in the tab list
+   */
+  inOverflow: false,
+  /**
+   * @property {int}
+   * int value representing the index at which the more tab starts (used to place the remaining tabs into the more tab)
+   */
+  tabMoreIndex: null,
+  /**
+   * @property {int}
+   * int value representing the index at which the default tab is located
+   */
+  defaultTabIndex: null,
+  /**
+   * @property {Array}
+   * Mapping of tab to the section
+   */
+  tabMapping: null,
+  /**
+   * @property {Array}
+   * Array holding the tab dom elements
+   */
+  tabs: null,
+
+  /**
+   * Changes the tab state in the tab list and changes visibility of content.
+   * @param {Object} The event type and source.
+   */
+  selectedTab: function selectedTab(params) {
+    const tab = params.$source;
```

Comment:
Should we add more formal methods like setCurrentTab and getCurrentTab and break out the logic? im not sure just think through what some one might what to hook into when working with tabs.